scripts/model-guard.sh

```
@@ -0,0 +1,110 @@
+# 4. If a model is hardcoded, resolves the correct one from providers.json
+# 5. If they differ, blocks with an error message showing the correct model
+#
+# BYPASS: Set OCTOPUS_MODEL_GUARD=off to disable (e.g., for debugging)
+# ═══════════════════════════════════════════════════════════════════════════════
+
+set -euo pipefail
+
+# Skip if guard is disabled
+[[ "${OCTOPUS_MODEL_GUARD:-on}" == "off" ]] && exit 0
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```

Shellcheck (0.11.0)

[warning] SCRIPT_DIR appears unused. Verify use (or export if used externally).

(SC2034)